src/ipc/ipc4/handler.c

```diff
@@ -487,9 +488,11 @@ static int ipc4_set_pipeline_state(struct ipc4_message_request *ipc4)
 	state.extension.dat = ipc4->extension.dat;
 	cmd = state.primary.r.ppl_state;
 
+#if !CONFIG_LIBRARY
```

Comment:
Can we minimise these ifdefs in the IPC i.e. can we have target specific static inlines that do the right thing for each target around mailbox. This would help readability.